src/t_set.c

```diff
@@ -237,28 +303,44 @@ unsigned long setTypeSize(const robj *subject) {
  * set. */
 void setTypeConvert(robj *setobj, int enc) {
     setTypeIterator *si;
-    serverAssertWithInfo(NULL,setobj,setobj->type == OBJ_SET &&
-                             setobj->encoding == OBJ_ENCODING_INTSET);
+    serverAssertWithInfo(NULL,setobj,setobj->type == OBJ_SET);
```

Comment:
I don't think we should remove this assertion, maybe we should change it to `(setobj->encoding == OBJ_ENCODING_INTSET || setobj->encoding == OBJ_ENCODING_LISTPACK) && setobj->encoding ! = enc`

Reply:
Since the code in this function now doesn't depend on setobj->encoding anymore, i.e. it uses only generic setType functions, I don't think it makes sense to assert on encoding. It just adds unnecessary logic.

Perhaps we can assert `(enc != setobj->encoding)`?

Reply:
`(enc != setobj->encoding)` sounds good.